src/main/java/de/tum/cit/aet/artemis/programming/service/sharing/SharingConnectorService.java

```diff
@@ -0,0 +1,292 @@
+     * @param installationName an optional descriptive name of the sharing application
+     * @return the sharing plugin config
+     */
+    public SharingPluginConfig getPluginConfig(URL apiBaseUrl, Optional<String> installationName) {
+        this.sharingApiBaseUrl = apiBaseUrl;
+        try {
+            this.installationName = installationName.or(() -> Optional.ofNullable(System.getenv("HOSTNAME"))).orElse(InetAddress.getLocalHost().getCanonicalHostName());
+        }
+        catch (UnknownHostException e) {
+            log.warn("Failed to determine hostname", e);
+            this.installationName = UNKNOWN_INSTALLATIONAME;
+        }
+        SharingPluginConfig.Action action = new SharingPluginConfig.Action("Import", "/sharing/import", actionName,
+                "metadata.format.stream().anyMatch(entry->entry=='artemis' || entry=='Artemis').get()");
+        lastHealthStati.add(new HealthStatus("Delivered Sharing Config Status to " + apiBaseUrl));
+        lastHealthStati.setLastConnect();
+        return new SharingPluginConfig("Artemis Sharing Connector", new SharingPluginConfig.Action[] { action });
+    }
+
+    /**
+     * just a maximum check length for validation limiting
+     */
+    private static final int MAX_API_KEY_LENGTH = 200;
+
+    /**
+     * Method used to validate the given authorizaion apiKey from Sharing
+     *
+     * @param apiKey the Key to validate
+     * @return true if valid, false otherwise
+     */
+    public boolean validate(String apiKey) {
+        if (apiKey == null || apiKey.length() > MAX_API_KEY_LENGTH) {
+            // this is just in case, somebody tries an attack
+            lastHealthStati.add(new HealthStatus("Failed api Key validation"));
+
+            return false;
+        }
+        Pattern p = Pattern.compile("Bearer\\s(.+)");
+        Matcher m = p.matcher(apiKey);
+        if (m.matches()) {
+            apiKey = m.group(1);
+        }
+
+        boolean success = Objects.equals(sharingApiKey, apiKey);
+        if (!success) {
+            lastHealthStati.add(new HealthStatus("Failed api Key validation (unequal)"));
+        }
+        return success;
+    }
+
+    /**
+     * At (spring) application startup, we request a reinitialization of the sharing platform .
+     * It starts a background thread in order not to block application startup.
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    public void triggerSharingReinitAfterApplicationStart() {
+        taskExecutor.execute(this::triggerReinit);
+    }
+
+    /**
+     * shuts down the service.
+     * currently just for test purposes
+     */
+    public void shutDown() {
+        sharingApiBaseUrl = null;
+        this.lastHealthStati.add(new HealthStatus("shutdown requested"));
+        this.lastHealthStati.resetLastConnect();
+    }
+
+    /**
+     * request a reinitialization of the sharing platform
+     */
+    public void triggerReinit() {
```

Comment:
```suggestion
    private void triggerReinit() {
```
can be private since only used by `triggerSharingReinitAfterApplicationStart`?